openedx/features/philu_courseware/views.py

```diff
@@ -51,13 +51,12 @@ def record_and_fetch_competency_assessment(request):
         score:Integer, 1
     :return: JSON
     """
-    competency_record = request.POST.dict()
-    competency_record['user'] = request.user.id
+    competency_record = request.data
 
-    serializer = CompetencyAssessmentRecordSerializer(data=competency_record)
+    serializer = CompetencyAssessmentRecordSerializer(data=competency_record, context=dict(request=request), many=True)
```

Comment:
Just curious why do we need to pass the context and then whole request as context, only case was `user` and we have set it to current user in Serializer.

Reply:
to user value in the serializer.

Reply:
Haven't we set it to current user in the serializer? Why do we again need to pass it? Also if need to pass a user then why are we passing the whole request object? why don't we just pass the use.

Reply:
@ahmed-faraz46 https://www.django-rest-framework.org/api-guide/validators/#currentuserdefault

Reply:
@zeshanarifios got it, Thanks!